mlflow/tensorflow.py

```diff
@@ -344,8 +345,15 @@ def predict(self, df):
                     for tensor_column_name in self.input_tensor_mapping.keys()
             }
             raw_preds = self.tf_sess.run(self.output_tensors, feed_dict=feed_dict)
-            pred_dict = {column_name: values.ravel() for column_name, values in raw_preds.items()}
-            return pandas.DataFrame(data=pred_dict)
+            # Output tensors can be multidimensional (e.g., softmax), so return in DataFrame with
+            # columns tensor0_0, tensor0_1, ..., tensor0_n, tensor1_0, tensor1_1, ..., tensor1_n
+            pred_df = pandas.DataFrame()
+            for output_name, output_values in raw_preds.items():
+                if output_values.ndim == 1:
+                    output_values = numpy.expand_dims(output_values, axis=1)
+                for column in range(output_values.shape[1]):
```

Comment:
Would this work as expected with ndim > 2?

Reply:
I think it makes most sense to raise an exception in the ndim > 2 case, especially if MLflow's design centers largely around tabular data structures (Pandas DataFrames, Spark DataFrames). This also ensures parity across the different deployment options (Spark UDF, SageMaker/Azure, local scoring server).

Another option would be reshaping to `[N, prod(dim2, ..., dimn)]` but it may be less confusing to just have users use a custom pyfunc at that point.

Reply:
I don't think we should throw an exception, it would make mlflow not work with any model which produces multidimensional output.